lib/create-subdomain/service.ts

```diff
@@ -80,56 +81,42 @@ export const createSubdomain = async ({
     calls.push(addressCallData);
   }
 
-  const dnsName = toHex(name),
-    owner = signerAddress;
+  const dnsName = toHex(packetToBytes(name));
 
-  let calldata;
-  try {
-    //L2 Resolver
-    // if the function registerParams exists, it's an l2 resolver, no error will be thrown
-    const ttl = 31556952000n;
-    const [value] = (await client.readContract({
-      address: resolverAddress,
-      abi: L1ResolverABI,
-      args: [dnsName, ttl],
-      functionName: "registerParams",
-    })) as [bigint, bigint, Hex];
-
-    const l2Calldata = {
-      functionName: "register",
-      abi: L1ResolverABI,
-      args: [
-        extractLabelFromName(name),
-        signerAddress, // owner
-        ttl,
-        `0x${"a".repeat(64)}` as Hex, // secret
-        "0xD9555FC98250d60974cf50cbf42Ffff361fE352e", // L2 resolver
-        calls, // calldata
-        false, // primaryName
-        0, // fuses
-        `0x${"a".repeat(64)}` as Hex,
-      ],
-
-      address: resolverAddress,
-      account: signerAddress,
-      value, // price
-    };
-    calldata = l2Calldata;
-  } catch (error) {
-    // Offchain Resolver
-    const ttl = 300;
-    const offChainCalldata = {
-      functionName: "register",
-      abi: DomainResolverABI,
-      args: [dnsName, ttl, owner, calls],
-      account: signerAddress,
-      address: resolverAddress,
-    };
-    console.error(error);
+  let value = 0n;
 
-    calldata = offChainCalldata;
+  try {
+    const [_value /* commitTime */ /* extraData */, ,] =
+      (await client.readContract({
+        address: resolverAddress,
+        abi: L1ResolverABI,
+        functionName: "registerParams",
+        args: [toHex(name), 31556952000n],
+      })) as [bigint, bigint, Hex];
+    value = _value;
+  } catch {
+    // interface not implemented by the resolver
   }
 
+  const calldata = {
+    functionName: "register",
+    abi: L1ResolverABI,
+    args: [
+      dnsName, // name
+      signerAddress, // owner
+      31556952000n,
```

Comment:
this should be the real duration

Reply:
some of these arguments could be the same as the ones used in [this function call](https://github.com/blockful-io/external-resolver-dapp/blob/e021293cea9db3fcd02e406c3a01becae7f08249/lib/utils/blockchain-txs.ts#L300)